### PR TITLE
DA-4648: Implement jira connector circle ci and docker file

### PR DESCRIPTION
**Pull request by ajinkyan83**
Signed-off-by: Ajinkya Nahar <ajinkyan@proximabiz.com>

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+          name: Push image
+          command: |
+            docker push $FULL_IMAGE_NAME
+  deploy_dev:
+    <<: *deploy
+    environment:
+      AWS_REGION_ENV_VAR: AWS_REGION_DEV
+      AWS_ACCESS_KEY_ID_ENV_VAR: AWS_ACCESS_KEY_ID_DEV
+      AWS_SECRET_ACCESS_KEY_ENV_VAR: AWS_SECRET_ACCESS_KEY_DEV
+      AWS_ACCOUNT_ID_ENV_VAR: "395594542180"
```
Comment (lukaszgryglicki) [inline .circleci/config.yml:65]: Hardcoded? Test & prod use env var...

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM alpine:3.14
+
+WORKDIR /app
+ENV JIRA_URL='<JIRA-URL>'
+ENV JIRA_USER='<JIRA-USER>'
+ENV JIRA_TOKEN='<JIRA-TOKEN>'
+ENV ES_URL='<ES-URL>'
+ENV STAGE='<STAGE>'
+RUN apk update && apk add git
```
Comment (lukaszgryglicki) [inline Dockerfile:9]: Do we need `git` command in Jira?

**Review by lukaszgryglicki — APPROVED** [review]
/lgtm
2 small comments.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+jobs:
+  build:
+    docker:
+      - image: circleci/golang:1.13-stretch-node
+    working_directory: /go/src/github.com/LF-Engineering/insights-connector-jira
+    steps:
+      - checkout
+      - run:
+          name: setup GITHUB token for private repositories
+          command: git config --global url."https://${GITHUB_TOKEN}:x-oauth-basic@github.com/LF-Engineering".insteadOf "https://github.com/LF-Engineering"
```
Comment (linonymous) [inline .circleci/config.yml:11]: if your go binary doesnt depend upon any private repositories in LFEngineering, you may get rid of this step & the GITHUB_TOKEN variable too.

**Review by linonymous — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (ajinkyan83) [reply .circleci/config.yml:62]: done

File: Dockerfile (hunk 2 above)
Comment (ajinkyan83) [reply Dockerfile]: fixed

File: .circleci/config.yml (hunk 3 above)
Comment (ajinkyan83) [reply .circleci/config.yml]: fixed.

**Pull request merged**